doc/Running and Writing Tests for Sia.md

```diff
@@ -0,0 +1,220 @@
+  * [Table-driven tests](#table)
+* [Questions?](#questions)
+
+<a name="existing"/>
+## Running tests for Sia
+Go's comprehensive [test package][pkg/testing] makes testing straightforward,
+particularly when you use the bundled tools included in the
+[Sia makefile][makefile], including `make test`, `make cover`, `make bench`,
+and their variants.
+
+<a name="update"/>
+### Updating code before testing
+If you just want to run existing tests on the codebase as is, you just need to
+pull the latest version of the upstream repo to your master branch.  (If that 
+sentence didn't make sense, go read
+[doc/Guide to Contributing to Sia.md][guide].)
+
+```bash
+# Make sure you are in the right directory.
+$ cd $GOPATH/src/github.com/<your Github username>/Sia
+# Also make sure you're working with the right branch.
+$ git checkout master
+$ git pull upstream master
+# Update `origin`, your fork of the repo.
+$ git push origin master
+```
+
+If you want to run tests on the new code you've added, first make sure the rest
+of the code is up to date. New code should be on its own branch (again, see
+[doc/Guide to Contributing to Sia.md][guide]).
+
+```bash
+# Make sure you are in the right directory.
+$ cd $GOPATH/src/github.com/<your Github username>/Sia
+# Checkout the branch you made the changes on.
+$ git checkout <branch name>
+# Stash any tracked but uncommitted changes.
+$ git stash
+# Then switch back to `master` and update it to match the original repo.
+$ git checkout master
+$ git pull upstream master
+# Update `origin`, your fork of the repo.
+$ git push origin master
+# Make the updated `master` the new base of the branch you made the changes on,
+# which involves reapplying all the commits made to that branch.  Without the
+# `--ignore-date` flag, git rebase changes the date on all the commits to the
+# current date.
+$ git checkout <branch name>
+$ git rebase master --ignore-date
+# Restore the changes you stashed earlier.
+$ git stash pop
+```
+When you call `rebase`, you may run into some merge conflicts.  Luke Champine's
+['How to into git and GitHub'][luke] has more details (and many useful tricks).
+
+Once the branch you want to test is up to date, you're ready to run some tests.
+
+<a name="entire"/>
+### Testing the entire build
+The `make test` command runs all tests (functions starting with `Test` in
+`_test.go` files) for each package, setting off a panic for any test that runs
+longer than 5s.  For verbose output, run `make test-v` (which panics after 15s
+instead of 5s).  Finally, `make test-long` has verbose output, only panics when
+a test takes 5 minutes, and also cleans up your code using `gofmt` and `golint`. 
+**You should run** `make test-long` **before each pull request.**
+
+Run `make cover` to run all tests for each package and generate color-coded
+.html visualizations of test coverage by function for each source file.  Open
+`cover/<module>.html` in a browser to inspect a module's test coverage.
```

Comment:
screenshot here would probably be useful as well, showing the rendered html of some covered code.